jitsi-utils/src/main/java/org/jitsi/utils/config/AbstractConfigProperty.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.utils.config;
+
+import org.jitsi.utils.config.strategy.prop_not_found.*;
+import org.jitsi.utils.config.strategy.read_frequency.*;
+
+import java.util.*;
+import java.util.function.*;
+
+/**
+ * A base helper class for modeling a configuration property.  Contains the
+ * code for iterating over multiple {@link Supplier}s for the first
+ * one which successfully returns a result; if none contain the property,
+ * it defers to the given {@link PropNotFoundStrategy}
+ *
+ * @param <T> the type of the configuration property's value
+ */
+public abstract class AbstractConfigProperty<T> implements ConfigProperty<T>
+{
+    protected final List<Supplier<T>> configValueSuppliers;
+    protected final ReadFrequencyStrategy<T> readFrequencyStrategy;
+    protected final PropNotFoundStrategy<T> propNotFoundStrategy;
+
+    public AbstractConfigProperty(PropertyConfig<T> builder)
+    {
+        builder.validate();
+        this.configValueSuppliers = builder.propValueSuppliers;
```

Comment:
Nit: I'm a little confused by the prefixes of the names here. Is there a difference between a "config" value supplier and a "prop" value supplier? Assuming there isn't, I think it'd be cleaner to just use `valueSuppliers` in both places.

Reply:
thanks, i agree.  fixed.